stackslib/src/chainstate/tests/runtime_tests.rs

```diff
@@ -0,0 +1,829 @@
+            type-checking before the contract is stored on-chain."
+        ),
+        BadBlockHeight(_) => Unreachable_Functionally(
+            "All block heights referenced via `at-block` or `get-block-info?` are guaranteed \
+            to exist in the node's historical database during normal execution. \
+            This error only surfaces if the chainstate is missing blocks or corrupted."
+        ),
+        NoSuchToken => Unreachable_Functionally(
+            "NFT operations return `none` when an instance does not exist. \
+            The `NoSuchToken` runtime error is only emitted from internal VM assertions \
+            and cannot be triggered by regular Clarity code unless storage is manually corrupted."
+        ),
+        NotImplemented => Unreachable_Functionally(
+            "Indicates use of an unimplemented VM feature. \
+            Can only be hit by directly invoking unfinished Rust internals – not reachable from Clarity."
+        ),
+        NoCallerInContext => Unreachable_Functionally(
+            "Every function call (public, private, or trait) is executed with a valid caller context. \
+            This error only appears when the execution environment is manually constructed incorrectly."
+        ),
+        NoSenderInContext => Unreachable_Functionally(
+            "Every on-chain transaction and contract-call has a well-defined sender. \
+            This error only occurs in malformed test harnesses."
+        ),
+        BadNameValue(_, _) => Unreachable_Functionally(
+            "Contract, function, trait, and variable names are fully validated during static analysis at publish time. \
+            The runtime only ever encounters already-validated names. \
+            Only corrupted state or manual VM manipulation can produce this error."
+        ),
+        UnknownBlockHeaderHash(_) => Tested(vec![unknown_block_header_hash_fork]),
+        BadBlockHash(_) => Tested(vec![bad_block_hash]),
+        UnwrapFailure => Tested(vec![unwrap_err_panic_on_ok_runtime, unwrap_panic_on_err_runtime]),
+        DefunctPoxContract => Tested(vec![defunct_pox_contracts]),
+        PoxAlreadyLocked => Ignored(
+            "The active PoX contract already returns ERR_STACKING_ALREADY_STACKED for double-locking attempts. \
+            The VM-level PoxAlreadyLocked error is only triggerable if locking occurs across PoX boundaries. \
+            This is better suited for unit testing."
+        ),
+        BlockTimeNotAvailable => Tested(vec![block_time_not_available]),
+    }
+}
+
+/// Error: [`RuntimeError::ArithmeticOverflow`]
+/// Caused by: overflow when doing `pow` arithmetic operation
+/// Outcome: block accepted.
+#[test]
+fn arithmetic_overflow_pow() {
+    contract_call_consensus_test!(
+        contract_name: "overflow-pow",
+        contract_code: &{
+        r#"
+    (define-public (trigger-overflow-pow)
+    (ok (pow 2 128))
+    )
+        "#
+        },
```

Comment:
nit: we can remove the `&{` and `r#` in here and all the other tests
```suggestion
        contract_code: "
    (define-public (trigger-overflow-pow)
        (ok (pow 2 128))
    )",
```